ethereum/light-client/src/client/checkpoint.rs

```diff
@@ -30,6 +31,34 @@ impl CheckpointClient {
         }
     }
 
+    /// Test the connection to the checkpoint provider.
+    ///
+    /// # Returns
+    ///
+    /// A result indicating whether the connection was successful.
+    pub(crate) async fn test_endpoint(&self) -> Result<(), ClientError> {
+        // Try to connect to the proof server
+        let mut retries = 0;
```

Comment:
Same as above on the backoff, and since this is the 3d snippet, we may want to generalize to a utility function.